Extensions/xkit_patches.js

```diff
@@ -184,6 +184,24 @@ XKit.extensions.xkit_patches = new Object({
 
 	patches: {
 		"7.9.1": function() {
+
+			XKit.post_listener.observer = new MutationObserver(mutations => {
+				const criteria = XKit.page.react ? "[data-id]" : ".post_container, .post";
+				const new_posts = mutations.find(({addedNodes}) => {
+					for (let i = 0; i < addedNodes.length; i++) {
+						const $addedNode = $(addedNodes[i]);
+						if ($addedNode.is(criteria) || $addedNode.find(criteria).length) {
+							return true;
+						}
+					}
+				}) !== undefined;
+
+				if (new_posts) {
+					const all_callbacks = [].concat.apply([], Object.values(XKit.post_listener.callbacks));
```

Comment:
This could use Array.from

Reply:
Oh wait no good heavens that is some structured data going on right there

Reply:
let's do `Object.values(XKit.postListener.callbacks).forEach(list => list.forEach(i => ... ) )`, the concat thing is cute but probably too cute.